packages/tpetra/core/compat/KokkosCompat_DefaultNode.hpp

```diff
@@ -0,0 +1,92 @@
+//@HEADER
+// ************************************************************************
+//
+//          Kokkos: Node API and Parallel Node Kernels
+//              Copyright (2008) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+
+#ifndef KOKKOSCOMPAT_DEFAULTNODE_HPP
+#define KOKKOSCOMPAT_DEFAULTNODE_HPP
+
+#include "TpetraCore_config.h"
+#include "KokkosClassic_DefaultNode_config.h"
+#include "KokkosCompat_ClassicNodeAPI_Wrapper.hpp"
+#include "Teuchos_RCP.hpp"
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
+namespace Teuchos {
+  // Dear users: This is just a forward declaration.
+  // Please skip over it.
+  class ParameterList;
+} // namespace Teuchos
+#endif // DOXYGEN_SHOULD_SKIP_THIS
+
+namespace KokkosClassic {
```

Comment:
Overall this PR is in the right direction.

Just making sure you understand that it would be better if you prefixed these symbols with `Tpetra`, `Tpetra_`, or nested it in the `Tpetra::` namespace because you technically do not own the `KokkosClassic` nor the `KokkosCompat` "brand".
The same apply to filenames and macros.
(It does not have to be `Tpetra` but it must be something you actually have ownership over.)

Reply:
@dalg24   A bunch of that KokkosClassic / KokkosCompat nomenclature predates Kokkos as a standalone thing.  Yay for technical debt :)

Moving to something like something like Tpetra_KokkosCompat_* for filenames and Tpetra::KokkosClassic for namespaces, would be better aligned with Kokkos' goals, yes?

Reply:
Yes